designdocs/schemas_protocols_intents.md

```diff
@@ -0,0 +1,445 @@
+| check_delegation_group       | `group_id: DelegationGroupId`<br/>`msa_id: MessageSourceId`<br/>`provider_id: ProviderId`<br/>`block_number: Option<BlockNumber>` | Returns the Intents currently-defined DelegationGroup, mapped to a boolean indicating the current delegation status of that Intent for the given MSA and Provider. |
+
+### 13. **Storage**<a id="storage"></a>
+
+#### Schemas
+
+The storage structures for Schemas will not fundamentally change, except for internal changes to the `SchemaInfo`
+structure that will require migration:
+
+* `payload_location`  & `settings` will migrate to `IntentInfo`
+* addition of `intent_id`
+
+#### Intents
+
+The <a href="#intent_struct">IntentInfo storage</a> will be as follows:
+
+```rust
+#[pallet::storage]
+pub(super) type IntentInfos<T: Config> =
+StorageMap<_, Twox64Concat, IntentId, IntentInfo, OptionQuery>;
+```
+
+#### Delegations
+
+There will be no change to Delegation storage; existing delegated `SchemaIds` will be interpreted as `IntentIds`.
+
+#### Delegation Groups
+
+The <a href="#delegation_group_struct">DelegationGroup structures</a> will be stored as follows:
+
+```rust
+#[pallet::storage]
+pub(super) type DelegationGroups<T: Config> =
+StorageMap<_, Twox64Concat, DelegationGroupId, DelegationGroup, OptionQuery>;
+```
+
+#### Name Registry
+
+The <a href="#name_registry_struct">Name Registry structures</a> will look as follows:
+
+```rust
+#[pallet::storage]
+pub(super) type NameRegistry<T: Config> = StorageDoubleMap<
+    _,
+    Blake2_128Concat,
+    ProtocolName,
+    Blake2_128Concat,
+    NameDescriptor,
+    RegisteredNameIdType,
+    ValueQuery,
+>;
+```
+
+#### `messages` pallet
+
+The `messages` pallet will require storage for an additional constant representing the block number at which this
+implementation is applied.
+
+```rust
+#[pallet::storage]
+pub type IntentBasedStorageCutoverBlockNumber<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
+```
+
+### Migrations
+
+Very little _existing_ data needs to be migrated; mostly just existing `SchemaInfo` storage. Anticipated migrations are
+as follows:
+
+1. Create a new `Intent` for every currently existing `Schema`, as follows:
+    1. The new `Intent` will have the same numeric ID value as the original Schema.
+    2. The new Intent shall inherit the `payload_location` and `settings` from the existing `SchemaInfo` object.
+2. Migrate existing schema name mappings as follows:
+    1. For each `SchemaNamespace` '\<protocol_name>'
+        1. For each `SchemaDescriptor` '\<name>' at index `n` belonging to a '\<protocol_name>'
+            * Create a new name mapping in the `NameRegistry` as `<protocol_name>.<name>_n` to `Intent(id)`
+3. Store the `messages` pallet cutover block number
```

Comment:
Optionally could seed the chain with some pre-defined Delegation Groups as part of the migration.